### PR TITLE
[#172762258] Alert when deselecting a favourite payment method

### PR DESCRIPTION
**Pull request by CrisTofani**
**Short description:**
This PR handles the action to remove a favourite payment method with an alert to explain to select another one in order to remove the previous selected method

**List of changes proposed in this pull request:**
- ts/screens/wallet/WalletsScreen.tsx
- locales/en/index.yml
- locales/it/index.yml

### CODE REVIEW

File: locales/it/index.yml [hunk 1]
```diff
@@ -550,6 +550,7 @@ wallet:
     titlePagoPaUpdateApp: Aggiornamento richiesto
     messagePagoPaUpdateApp: "IO introduce spesso piccole migliorie e nuove funzioni: per continuare ad usare il portafoglio è necessario aggiornare l'applicazione all'ultima versione."
     btnUpdateApp: Aggiorna l'app IO
+    favourite: Non è possibile rimuovere un metodo di pagamento preferito, per modificarlo selezionane un altro
```
Comment (Undermaken) [inline locales/it/index.yml:553]:
```suggestion
    favourite: Il metodo selezionato, essendo l'unico, è anche il preferito. Per impostare un altro metodo preferito aggiungilo e fai tap sulla stellina.
```
Could it be more clear?

File: ts/screens/wallet/WalletsScreen.tsx [hunk 2]
```diff
@@ -84,9 +85,11 @@ class WalletsScreen extends React.Component<Props> {
         wallet={item}
         isFavorite={isFavorite}
         onSetFavorite={(willBeFavorite: boolean) =>
-          this.props.setFavoriteWallet(
-            willBeFavorite ? item.idWallet : undefined
-          )
+          willBeFavorite
```
Comment (Undermaken) [inline ts/screens/wallet/WalletsScreen.tsx:88]: It's the same but usually multi ternary-operator could be hard to read

```typescript
onSetFavorite={(willBeFavorite: boolean) =>
          fromPredicate(wbf => wbf === true)(willBeFavorite).foldL(
            () => {
              if (this.props.wallets.length > 0) {
                Alert.alert(I18n.t("wallet.alert.favourite"));
              }
            },
            () => this.props.setFavoriteWallet(item.idWallet)
          )
        }
```

File: ts/screens/wallet/WalletsScreen.tsx [hunk 3]
```diff
@@ -84,9 +85,11 @@ class WalletsScreen extends React.Component<Props> {
...
+            ? this.props.setFavoriteWallet(item.idWallet)
+            : this.props.wallets.length > 0
```
Comment (Undermaken) [inline ts/screens/wallet/WalletsScreen.tsx:90]:
```suggestion
            : this.props.wallets.length > 1
```
Should we show an alert only if the user set another wallet as favourite?

Comment (Undermaken) [reply ts/screens/wallet/WalletsScreen.tsx:90]: PS: last commit on [io-dev-server ](https://github.com/pagopa/io-dev-api-server)now supports set wallet as favourite

Comment (CrisTofani) [reply ts/screens/wallet/WalletsScreen.tsx]: @Undermaken If we adopt this condition we should avoid the click on the star icon when we have only one method available.
At the moment we show the alert anytime the user's trying to remove a favourite method

**Pull request merged**